js/phemex.js

```diff
@@ -2794,6 +2807,94 @@ module.exports = class phemex extends Exchange {
         return await this.privatePutPositionsLeverage (this.extend (request, params));
     }
 
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const accountsByType = this.safeValue (this.options, 'accountsByType', {});
+        const fromId = this.safeString (accountsByType, fromAccount);
+        const toId = this.safeString (accountsByType, toAccount);
+        let direction = undefined;
+        if (fromId === undefined) {
+            const keys = Object.keys (accountsByType);
+            throw new ExchangeError (this.id + ' fromAccount must be one of ' + keys.join (', '));
+        }
+        if (toId === undefined) {
+            const keys = Object.keys (accountsByType);
+            throw new ExchangeError (this.id + ' toAccount must be one of ' + keys.join (', '));
+        }
+        if (fromId === 'spot' && toId === 'future') {
+            direction = '2';
+        }
+        if (fromId === 'future' && toId === 'spot') {
+            direction = '1';
+        }
+        if (direction === undefined) {
+            throw new ExchangeError (this.id + ' transfer can only be down from future to spot or from spot to future');
+        }
+        const scaledAmmount = this.toEv (amount, currency);
+        const request = {
+            'currency': currency['id'],
+            'moveOp': direction,
+            'amountEv': scaledAmmount,
+        };
+        const response = await this.privatePostAssetsTransfer (this.extend (request, params));
+        //
+        //     {
+        //         code: '0',
+        //         msg: 'OK',
+        //         data: {
+        //             linkKey: '8564eba4-c9ec-49d6-9b8c-2ec5001a0fb9',
+        //             userId: '4018340',
+        //             currency: 'USD',
+        //             amountEv: '10',
+        //             side: '2',
+        //             status: '10'
+        //         }
+        //     }
+        //
+        const data = this.safeValue (response, 'data', {});
+        return this.parseTransfer (data, currency);
+    }
+
+    parseTransfer (transfer, currency = undefined) {
+        const id = this.safeString (transfer, 'linkKey');
+        const status = this.safeString (transfer, 'status');
+        const amountEv = this.safeString (transfer, 'amountEv');
+        const amountTransfered = this.fromEv (amountEv, currency);
+        const code = this.safeString (currency, 'code');
```

Comment:
The `currency` can be `undefined` in this method, therefore we have to use `safeCurrency / safeCurrencyCode` just like we did with `safeMarket / safeSymbol` in fee-fetching methods.

Reply:
good catch

Reply:
I saw `safeCurrencyCode` doesn't protect for both `currencyId = undefined && currency = undefined`, so I added manually a check before calling it